ts/features/bonusVacanze/screens/CheckBonusEligibilityScreen.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { RTron } from "../../../boot/configureStoreAndPersistor";
+import I18n from "../../../i18n";
+import { GlobalState } from "../../../store/reducers/types";
+import { GenericLoadingErrorScreen } from "../components/GenericLoadingErrorScreen";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+const loadingCaption = I18n.t(
+  "bonus.bonusVacanza.checkBonusEligibility.loading"
+);
+/**
+ * This component link the generic {@link GenericLoadingErrorScreen} with the application flow using the {@link connect}
+ * of redux.
+ * @param props
+ * @constructor
+ */
+const CheckBonusEligibilityScreen: React.FunctionComponent<Props> = props => {
+  return (
+    <GenericLoadingErrorScreen
+      {...props}
+      loadingCaption={loadingCaption}
+      loadingOpacity={1}
+    />
+  );
+};
+
+const mapDispatchToProps = (_: Dispatch) => ({
+  // TODO: link with the right dispatch action
+  onCancel: () => {
+    RTron.log("CANCEL");
+  },
+  // TODO: link with the right dispatch action
+  onRetry: () => {
+    RTron.log("RETRY");
+  }
+});
+
+const mapStateToProps = (_: GlobalState) => ({
+  // TODO: link with the right reducer
+  isLoading: true,
+  // TODO: link with the right reducer
+  errorText: "A text that explains why the operation failed"
```

Comment:
I guess the error could be a network error, otherwise we should receive a response from our backend.
Could it be a solution to show an error toast informing the user that something went wrong?

Reply:
If we want to have a generic error, I can remove the `errorText` and use the default message used within the `GenericErrorComponent` ("Ooops, qualcosa è andato storto! Probabilmente è un errore momentaneo, riprova per piacere!"). In this case we can also avoid a toast that which would repeat the error information. What you think?

Reply:
Sounds good!